.travis.yml

```diff
@@ -199,7 +201,8 @@ script:
   - echo "Configure flags are $CONFIGURE_FLAGS"
   - ./configure $CONFIGURE_FLAGS
   ## We run `make check` because that's what https://jenkins.torproject.org does.
-  - if [[ "$DISTCHECK" == "" ]]; then make check; fi
+  - if [[ "$DISTCHECK" == "" && "$TEST_STEM" == "" ]]; then make check; fi
+  - if [[ "$TEST_STEM" != "" ]]; then git clone --depth 1 https://github.com/torproject/stem.git ; export STEM_SOURCE_DIR=`pwd`/stem; make src/app/tor test-stem; fi
```

Comment:
Technically, git clone stem belongs in the install section.
We also do our environmental variable setup in install.
Please also log the stem version. I think git clone already logs the stem commit.

It seems like distcheck and test-stem are incompatible, because distcheck deletes src/app/tor.
If both options are specified, should we skip stem and log a message?

Nitpick: I'd like to see test stem added as the last item in both sections, but it doesn't really matter.

Reply:
These are addressed in further commits now. Opened a new pull request with rebased branch and cleaned up git history: https://github.com/torproject/tor/pull/423